### Understand the Data

In [28]:
import pandas as pd 

df_movies = pd.read_csv('movies.csv', header = 0)
df_ps = pd.read_csv('past_screenings.csv', header = 0)


In [29]:
df_movies.head()

id               movie    year      genre  duration        country  \
0   71      Charlie Muffin  1979.0        Spy     109.0  Great Britain   
1   78  Absolute Beginners  1986.0    Musical     100.0  Great Britain   
2   94     Arrêt sur image  1987.0      Drama      77.0         France   
3   97               Water  1985.0  Grotesque      97.0  Great Britain   
4  109    Adam and Evelyne  1949.0     Comedy      92.0  Great Britain   

          director  humor  rhythm  effort  tension  vote  avg_vote  
0        Jack Gold    1.0     2.0     1.0      3.0   6.0      13.0  
1    Julien Temple    1.0     2.0     0.0      2.0   6.0      31.0  
2  Christine Allen    0.0     1.0     1.0      0.0   4.0       4.0  
3     Dick Clement    2.0     3.0     0.0      1.0   6.0       5.0  
4    Harold French    1.0     2.0     1.0      0.0   6.0      11.0

In [30]:
df_ps.head()

week  day  room  screening                        movie
0     0    0     0          0           Wallander: Skytten
1     0    0     0          1                       Canola
2     0    0     0          2                  Guns Akimbo
3     0    0     1          3  La mano de un hombre muerto
4     0    0     1          4       Vendetta for the Saint

In [31]:
df_movies.info()
df_ps.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5794 entries, 0 to 5793
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        5794 non-null   object 
 1   movie     5793 non-null   object 
 2   year      5793 non-null   float64
 3   genre     5780 non-null   object 
 4   duration  5793 non-null   float64
 5   country   5793 non-null   object 
 6   director  5793 non-null   object 
 7   humor     5793 non-null   float64
 8   rhythm    5793 non-null   float64
 9   effort    5793 non-null   float64
 10  tension   5793 non-null   float64
 11  vote      5793 non-null   float64
 12  avg_vote  5793 non-null   float64
dtypes: float64(8), object(5)
memory usage: 588.6+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20000 entries, 0 to 19999
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   week       20000 non-null  int64 
 1   day        20000 non-nul

In [32]:
df_ps.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20000 entries, 0 to 19999
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   week       20000 non-null  int64 
 1   day        20000 non-null  int64 
 2   room       20000 non-null  int64 
 3   screening  20000 non-null  int64 
 4   movie      20000 non-null  object
dtypes: int64(4), object(1)
memory usage: 781.4+ KB


## Clean the Data

##### Détection d observations identiques : code 

In [33]:
dp_movie = df_movies.duplicated()
dp_ps = df_ps.duplicated()

In [34]:
dp_movie.any() # no duplicates

np.False_

In [35]:
dp_ps.any() #no duplicates

np.False_

In [36]:
df_movies[dp_movie] #no duplicates
df_ps[dp_ps] #no duplicates

Empty DataFrame
Columns: [week, day, room, screening, movie]
Index: []

##### Vérifier les valeurs manquantes

In [37]:
df_movies.isnull().sum() 

id           0
movie        1
year         1
genre       14
duration     1
country      1
director     1
humor        1
rhythm       1
effort       1
tension      1
vote         1
avg_vote     1
dtype: int64

In [38]:
df_ps.isnull().sum()

week         0
day          0
room         0
screening    0
movie        0
dtype: int64

In [39]:
missing_rows = df_movies[df_movies.isnull().any(axis=1)]
missing_rows

id  \
1745                                              23892   
1793                                              24792   
2502                                              36267   
2563                                              37261   
2586                                              37724   
2652                                              38479   
2781                                              40007   
2902                                              41265   
2973                                              42145   
3017                                              42646   
3083                                              43546   
3120                                              43940   
3361  49576,"""Sherlock"" The Great Game",2010,Thril...   
5267                                             175367   

                                           movie    year genre  duration  \
1745                          Monsieur Batignole  2002.0   NaN     100.0   
1793                                 Bord de mer  2002.0   NaN      90.0   
2502                               Pumhaeng zero  2002.0   NaN      99.0   
2563                      En la ciudad de Sylvia  2007.0   NaN      90.0   
2586                            A vot' bon coeur  2004.0   NaN      93.0   
2652                                De la guerre  2007.0   NaN     130.0   
2781                                       Ander  2009.0   NaN     128.0   
2902      Bancs publics (Versailles rive droite)  2009.0   NaN     110.0   
2973                           L'arroseur arrosé  1897.0   NaN      49.0   
3017  Histoire(s) du cinéma: Une histoire seule   1989.0   NaN      42.0   
3083                                Snabba Cash   2010.0   NaN     124.0   
3120         Almanya - Willkommen in Deutschland  2011.0   NaN      97.0   
3361                                         NaN     NaN   NaN       NaN   
5267                           House of Whipcord  1974.0   NaN     102.0   

            country            director  humor  rhythm  effort  tension  vote  \
1745         France       Gérard Jugnot    2.0     3.0     4.0      3.0   7.0   
1793         France  Julie Lopes-Curval    3.0     3.0     2.0      2.0   6.0   
2502    South Korea        Jo Geun-shik    1.0     2.0     1.0      1.0   6.0   
2563          Spain    José Luis Guerín    0.0     0.0     0.0      0.0   3.0   
2586         France      Paul Vecchiali    2.0     2.0     1.0      2.0   6.0   
2652         France    Bertrand Bonello    2.0     2.0     3.0      2.0   7.0   
2781          Spain      Roberto Castón    0.0     0.0     0.0      0.0   7.0   
2902         France     Bruno Podalydès    0.0     0.0     0.0      0.0   6.0   
2973         France           Alice Guy    0.0     0.0     0.0      0.0   6.0   
3017         France     Jean-Luc Godard    0.0     0.0     0.0      0.0   8.0   
3083         Sweden     Daniel Espinosa    0.0     0.0     0.0      0.0   5.0   
3120        Germany   Yasemin Samdereli    3.0     3.0     3.0      2.0   6.0   
3361            NaN                 NaN    NaN     NaN     NaN      NaN   NaN   
5267  Great Britain         Pete Walker    0.0     0.0     0.0      0.0   7.0   

      avg_vote  
1745      37.0  
1793      11.0  
2502       3.0  
2563       4.0  
2586       8.0  
2652      13.0  
2781      14.0  
2902       3.0  
2973      12.0  
3017       7.0  
3083       4.0  
3120      51.0  
3361       NaN  
5267       3.0

In [40]:
df_movies.loc[1745, 'genre'] = "Comedy"
df_movies.loc[1745]

id                       23892
movie       Monsieur Batignole
year                    2002.0
genre                   Comedy
duration                 100.0
country                 France
director         Gérard Jugnot
humor                      2.0
rhythm                     3.0
effort                     4.0
tension                    3.0
vote                       7.0
avg_vote                  37.0
Name: 1745, dtype: object

In [41]:
df_movies.loc[1793, 'genre'] = "Drama"
df_movies.loc[1793]

id                       24792
movie              Bord de mer
year                    2002.0
genre                    Drama
duration                  90.0
country                 France
director    Julie Lopes-Curval
humor                      3.0
rhythm                     3.0
effort                     2.0
tension                    2.0
vote                       6.0
avg_vote                  11.0
Name: 1793, dtype: object

In [43]:
df_movies.loc[2502, 'genre'] = "Comedy"
df_movies.loc[2502]

id                  36267
movie       Pumhaeng zero
year               2002.0
genre              Comedy
duration             99.0
country       South Korea
director     Jo Geun-shik
humor                 1.0
rhythm                2.0
effort                1.0
tension               1.0
vote                  6.0
avg_vote              3.0
Name: 2502, dtype: object

In [44]:
# Afficher les colonnes du DataFrame
print(df_movies.columns)

Index(['id', 'movie', 'year', 'genre', 'duration', 'country', 'director',
       'humor', 'rhythm', 'effort', 'tension', 'vote', 'avg_vote'],
      dtype='object')


In [45]:
if 'date' in df_movies.columns:
	df_movies = df_movies.drop('date', axis=1)
else:
	print("Column 'date' does not exist in the dataframe.")

Column 'date' does not exist in the dataframe.


In [49]:
df_movies.loc[2502]

id                  36267
movie       Pumhaeng zero
year               2002.0
genre              Comedy
duration             99.0
country       South Korea
director     Jo Geun-shik
humor                 1.0
rhythm                2.0
effort                1.0
tension               1.0
vote                  6.0
avg_vote              3.0
Name: 2502, dtype: object

In [50]:
df_movies.loc[2563, 'genre'] = "Drama"
df_movies.loc[2563]

id                           37261
movie       En la ciudad de Sylvia
year                        2007.0
genre                        Drama
duration                      90.0
country                      Spain
director          José Luis Guerín
humor                          0.0
rhythm                         0.0
effort                         0.0
tension                        0.0
vote                           3.0
avg_vote                       4.0
Name: 2563, dtype: object

In [51]:
df_movies.loc[2586, 'genre'] = "Comedy"
df_movies.loc[2586]

id                     37724
movie       A vot' bon coeur
year                  2004.0
genre                 Comedy
duration                93.0
country               France
director      Paul Vecchiali
humor                    2.0
rhythm                   2.0
effort                   1.0
tension                  2.0
vote                     6.0
avg_vote                 8.0
Name: 2586, dtype: object

In [52]:
df_movies.loc[2652, 'genre'] = "Drama"
df_movies.loc[2652]	

id                     38479
movie           De la guerre
year                  2007.0
genre                  Drama
duration               130.0
country               France
director    Bertrand Bonello
humor                    2.0
rhythm                   2.0
effort                   3.0
tension                  2.0
vote                     7.0
avg_vote                13.0
Name: 2652, dtype: object

In [53]:
df_movies.loc[2781, 'genre'] = "Drama"
df_movies.loc[2781]	

id                   40007
movie                Ander
year                2009.0
genre                Drama
duration             128.0
country              Spain
director    Roberto Castón
humor                  0.0
rhythm                 0.0
effort                 0.0
tension                0.0
vote                   7.0
avg_vote              14.0
Name: 2781, dtype: object

In [54]:
df_movies.loc[2902, 'genre'] = "Drama"
df_movies.loc[2902]	

id                                           41265
movie       Bancs publics (Versailles rive droite)
year                                        2009.0
genre                                        Drama
duration                                     110.0
country                                     France
director                           Bruno Podalydès
humor                                          0.0
rhythm                                         0.0
effort                                         0.0
tension                                        0.0
vote                                           6.0
avg_vote                                       3.0
Name: 2902, dtype: object

In [55]:
df_movies.loc[2973, 'genre'] = "Comedy"
df_movies.loc[2973]

id                      42145
movie       L'arroseur arrosé
year                   1897.0
genre                  Comedy
duration                 49.0
country                France
director            Alice Guy
humor                     0.0
rhythm                    0.0
effort                    0.0
tension                   0.0
vote                      6.0
avg_vote                 12.0
Name: 2973, dtype: object

In [56]:
df_movies.loc[3017, 'genre'] = "Documentary"
df_movies.loc[3017]

id                                               42646
movie       Histoire(s) du cinéma: Une histoire seule 
year                                            1989.0
genre                                      Documentary
duration                                          42.0
country                                         France
director                               Jean-Luc Godard
humor                                              0.0
rhythm                                             0.0
effort                                             0.0
tension                                            0.0
vote                                               8.0
avg_vote                                           7.0
Name: 3017, dtype: object

In [57]:
df_movies.loc[3083, 'genre'] = "Crime"
df_movies.loc[3083]


id                    43546
movie          Snabba Cash 
year                 2010.0
genre                 Crime
duration              124.0
country              Sweden
director    Daniel Espinosa
humor                   0.0
rhythm                  0.0
effort                  0.0
tension                 0.0
vote                    5.0
avg_vote                4.0
Name: 3083, dtype: object

In [58]:
df_movies.loc[3120, 'genre'] = "Comedy"
df_movies.loc[3120]

id                                        43940
movie       Almanya - Willkommen in Deutschland
year                                     2011.0
genre                                    Comedy
duration                                   97.0
country                                 Germany
director                      Yasemin Samdereli
humor                                       3.0
rhythm                                      3.0
effort                                      3.0
tension                                     2.0
vote                                        6.0
avg_vote                                   51.0
Name: 3120, dtype: object

In [59]:
df_movies.loc[5267, 'genre'] = "Thriller"
df_movies.loc[5267]

id                     175367
movie       House of Whipcord
year                   1974.0
genre                Thriller
duration                102.0
country         Great Britain
director          Pete Walker
humor                     0.0
rhythm                    0.0
effort                    0.0
tension                   0.0
vote                      7.0
avg_vote                  3.0
Name: 5267, dtype: object

In [61]:
df_movies.isnull().sum() 

id          0
movie       1
year        1
genre       1
duration    1
country     1
director    1
humor       1
rhythm      1
effort      1
tension     1
vote        1
avg_vote    1
dtype: int64

In [62]:
missing_rows = df_movies[df_movies.isnull().any(axis=1)]
missing_rows

id movie  year genre  \
3361  49576,"""Sherlock"" The Great Game",2010,Thril...   NaN   NaN   NaN   

      duration country director  humor  rhythm  effort  tension  vote  \
3361       NaN     NaN      NaN    NaN     NaN     NaN      NaN   NaN   

      avg_vote  
3361       NaN

##### Supprimer les lignes où les colonnes critiques ont des valeurs manquantes

In [63]:
df_movies_cleaned = df_movies.dropna()

In [64]:
len(df_movies_cleaned)

5793

On est passé de 5794 lignes à 5793 lignes (5796 - 1).
Cela représente une perte de 0,01726 % des données, ce qui est négligeable dans la plupart des cas.